de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsVerificationBuilder.java

```diff
@@ -0,0 +1,155 @@
+			
+			if (resourceSet != null && !resourceSet.isOpen()) {
+				Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VerificationBuilder: Exited since project is closed!"));
+				return;
+			}
+			
+			delta.accept(new IResourceDeltaVisitor() {
+				public boolean visit(IResourceDelta delta) {
+					IResource iResource = delta.getResource();
+				
+					if (iResource instanceof IFile) {
+						// We should only process DVLM resources in terms of model updates
+						IFile iFile = (IFile) iResource;
+												
+						if (!iFile.exists() || !VirSatProjectCommons.isDvlmFile(iFile)) {
+							return true;
+						}
+
+						Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(),
+								"VerificationBuilder: Processing incremental build on resource (" + iResource + ")"));
+						
+						Resource resource = resourceSet.safeGetResource(iFile, false);
+						if (resource.getContents().get(0) instanceof StructuralElementInstance) {
+							buildSei(new BeanStructuralElementInstance((StructuralElementInstance) resource.getContents().get(0)), monitor);
+						}
+						
+					}
+
+					return true; // visit children too
+				}
+			});
+			
+		} catch (CoreException e) {
+			Activator.getDefault().getLog().log(new Status(Status.ERROR, Activator.getPluginId(), Status.ERROR, "VerificationBuilder: Received an exception", e));
+		}
+		
+		Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VerificationBuilder: Finished incremental build!"));
+	}
+
+	@Override
+	protected void fullBuild(IProgressMonitor monitor) {
+		Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VerificationBuilder: Starting full build!"));
+		VirSatResourceSet resourceSet = getResourceSet();
+		if (resourceSet != null && !resourceSet.isOpen()) {
+			Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VerificationBuilder: Exited since project is closed!"));
+			return;
+		}
+		
+		Set<StructuralElementInstance> seis = getResourceSet().getAllSeisInProject();
+		
+		SubMonitor subMonitor = SubMonitor.convert(monitor, seis.size());
+		subMonitor.beginTask("Executing Verification of SEIs", seis.size());
+		
+		for (StructuralElementInstance sei : seis) {
+			BeanStructuralElementInstance bean = new BeanStructuralElementInstance(sei);
+			buildSei(bean, subMonitor);
+			subMonitor.worked(1);
+		}
+		
+		
+		Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "VerificationBuilder: Finished full build!"));
+		
+	}
+	
+	/**
+	 * Run the verification on a SEI within the project
+	 * @param seiBean the SEI Bean
+	 * @param monitor the progress monitor
+	 */
+	protected void buildSei(BeanStructuralElementInstance seiBean, IProgressMonitor monitor) {
+		List<CategoryAssignment> caSpecifications = seiBean.getStructuralElementInstance().getCategoryAssignments().stream().

```

Comment:
Any reason we cannot do `List<Requirementsspecification> specifications = seiBean.getAll(RquirementsSpecification.class)`